src/BCReader.jl

```diff
@@ -0,0 +1,309 @@
+#=
+    BCReader
+
+This module coordinates reading of boundary conditions from NetCDF files,
+as well as regridding calls and temporal interpolations from
+monthly to daily intervals.
+=#
+module BCReader
+
+using ..Utilities, ..Regridder, ..CallbackManager
+using ClimaCore: Fields
+using ClimaComms
+using Dates
+using JLD2
+
+export BCFileInfo, bcfile_info_init, update_midmonth_data!, next_date_in_file, interpolate_midmonth_to_daily
+
+
+"""
+    BCFileInfo
+
+Stores information specific to each boundary condition from a file and each variable.
+
+# Inputs
+    FT::F                           # float type
```

Comment:
If you use Charlie's suggestion, remember to update the docstring :)

Reply:
Will do!